tests/test_unit/test_file_utils.py

```diff
@@ -0,0 +1,196 @@
+import shutil
+from pathlib import Path
+
+import h5py
+import numpy as np
+import pytest
+
+from brainglobe_stitch.file_utils import (
+    check_mesospim_directory,
+    create_pyramid_bdv_h5,
+    get_big_stitcher_transforms,
+    get_slice_attributes,
+    parse_mesospim_metadata,
+)
+
+TEMP_DIR = Path("./temp_directory")
+NUM_RESOLUTIONS = 5
+NUM_TILES = 8
+CHANNELS = ["561 nm", "647 nm"]
+PIXEL_SIZE_XY = 4.08
+PIXEL_SIZE_Z = 5.0
+EXPECTED_TRANSFORMS = np.array(
+    [
+        [2, 130, 4, 132, 3, 113],
+        [0, 128, 120, 248, 2, 112],
+        [2, 130, 4, 132, 3, 113],
+        [0, 128, 120, 248, 2, 112],
+        [118, 246, 7, 135, 6, 116],
+        [116, 244, 123, 251, 5, 115],
+        [118, 246, 7, 135, 6, 116],
+        [116, 244, 123, 251, 5, 115],
+    ]
+)
+
+
+@pytest.fixture
+def bad_bdv_directory():
+    bad_dir = Path("./bad_directory")
+    bad_dir.mkdir()
+
+    yield bad_dir
+
+    shutil.rmtree(bad_dir)
+
+
+def test_create_pyramid_bdv_h5(naive_bdv_directory, test_data_directory):
```

Comment:
Due to the large size of the files some of the functions alter the input files. `test_data_directory` contains a clean version of the test data that's never modified, `naive_bdv_directory` is generated by copying the clean version to a separate directory, this is the dirty copy that gets cleaned up after every test. 

This test uses the standard structure of the h5 file to test the `create_pyramid_function`. The h5 file has a dataset for each tile that stores the resolution and subdivision, those should have shape (pyramid_depth, 3), the root also has a `t00000` dataset that contains the actual image data for each slice. The full resolution would be stored in `t00000/s00/0/cells`, the first level of the resolution pyramid is stored at `t00000/s00/1/cells`, etc...

This test first does a sanity check that the h5 file in the `naive_bdv_directory` directory starts with no resolution pyramid, i.e. the size of the first axis of `s00/resolutions` and  `s00/subdivisions` should be 1 for all tiles, and the only group stored in `t00000/s00/` should be `0`. The next call loops through the `create_pyramid_bdv_h5` functions as it's a generator and checks that the `progress` value yielded is as expected (percent complete). The last part then compares the altered file with the original file in the clean `test_data_directory` to make sure no tiles were deleted, and then checks that the expected number of resolutions were created in the resolution pyramid.

It's a lengthy test, but as the test requires copying some data, I didn't want to split it into sub-components to keep test time low. Ideally I would split the `create_pyramid_bdv_h5` function into it's components, test each individually, and then do a quick "integration" test on the full function. Something to come back to in a refactor potentially! I'm not happy with the performance of this function at the moment so it may get rewritten completely.

Reply:
I'll make sure to document this better in the docstring!

Reply:
Thanks @IgorTatarnikov that makes sense! I think the test length is OK and it is in general clear, the only thing that didn't click for me was that `create_pyramid_function` was alerting the file in `test_data_directory` which is then reloaded and checked. Like you said a small explanation of this in a docstring and I think the test would be clear!